### ISSUE TITLE
Query's stream ends before listing all documents

### ISSUE THREAD

**Issue by elbajo** [googleapis/python-firestore#99]
### Environment details

  - OS type and version: MacOS / Debian
  - Python version: Python 3.7.6
  - pip version: 20.1.1
  - `google-cloud-firestore` version: 1.8.1

#### Steps to reproduce

Query a large enough collection and update each document. The query's stream will end before reaching the end of the collection.
In my case collection is ~3500 and the stream always stops once ~400 elements have been updated.
Note that if the first 1000 elements have already been updated it will go through the first 1000 without stopping and then stop after ~400 new elements have been updated.

This makes me think that maybe my query is expiring after X seconds which stops the stream? If so it would be great to get an error instead of stopping silently.

I tried adding time.sleep(0.5) inside the stream's loop and the stream stopped after 90 elements instead of 400 which seem to confirm the timeout after X seconds theory.

It also seems like the query costs me the entire 3500 reads each time even if it stops after 400 elements...

I probably could go through all the documents first and then update them one by one but I feel like I shouldn't need to do that.

I tried this mostly on MacOS but also tried on Debian where I got similar results (it seems to run faster there so the stream only stopped after 700 elements)

This sounds like a very straightforward use case so I feel like I must be doing something wrong?

#### Code example

```python
places_query = db.collection(u'places')
i = 0
for place in place_query.stream():
  i += 1
  print(f'Document #{i}')
  place.reference.update({'new_field': 'new_value'})
```

#### Stack trace

No stacktrace... stream just stops.

**Comment by elbajo** [comment]
Oh... just found https://github.com/googleapis/google-cloud-python/issues/6033 which is probably my issue.
TLDR is: queries timeout after `max_rpc_timeout_millis` is reached which is set to 30 seconds

Still think it would be nice to have some kind of error instead of silently failing, this was a painful issue to debug.

**Comment by crwilcox** [comment]
@elbajo thank you for raising this. I am going to mark this as a bug and keep it open. We are in the process of authoring a v2 version with substantial changes tot he gRPC portions of this library. I think it makes sense to followup at that point again.

**Comment by crwilcox** [comment]
@elbajo curious if you have adopted our v2 and are still encountering this issue?
